serve/package.go

```diff
@@ -455,6 +462,7 @@ func (s *PluginServe) newCmdPluginPackage() *cobra.Command {
 	cmd.Flags().StringP("dist-dir", "D", "", "dist directory to output the built plugin. (default: <plugin_directory>/dist)")
 	cmd.Flags().StringP("docs-dir", "", "", "docs directory containing markdown files to copy to the dist directory. (default: <plugin_directory>/docs)")
 	cmd.Flags().StringP("message", "m", "", "message that summarizes what is new or changed in this version. Use @<file> to read from file. Supports markdown.")
+	cmd.Flags().StringP("build-tags", "", "", "build tags to use when building the plugin. (default: grpcnotrace)")
```

Comment:
This should not be a CLI flag, it should be a field in `BuildTarget` https://github.com/cloudquery/plugin-sdk/blob/b27291d68e3dca2019a36278f704c1a1f9263cd8/plugin/plugin_package.go#L39 controlled via `WithBuildTargets` https://github.com/cloudquery/plugin-sdk/blob/b27291d68e3dca2019a36278f704c1a1f9263cd8/plugin/options.go#L20

We can change the default in https://github.com/cloudquery/plugin-sdk/blob/b27291d68e3dca2019a36278f704c1a1f9263cd8/plugin/plugin_package.go#L80 to include `grpcnotrace`

Reply:
If we do it there then as soon as we use a go tag (eg `-tags=fipsEnabled`) then it will be overridden.

Reply:
Can't the plugin set multiple tags? i.e. use the default build targets, then append to the default tags the `fipsEnabled` one?

Reply:
Using a CLI flag is problematic since it's not backwards compatible, i.e. older plugins will fail is you pass them `--build-tags`

Reply:
Multiple tags is difficult because go requires a single `-tags` and doesn't combine between `GOFLAGS` and `-tags` it will just take the last one.

This is already a breaking change...

Reply:
This PR also overrides the tags from the flag so won't the plugin will need to be called with `--buildTags=grpcnotrace,<additional-tags>` by whoever is invoking the package command?

I think the cleaner approach would be for the plugin to decide on the build tags, not the caller of the package command.

For FIPS plugins we would create a `plugin_fips.go` similar to https://github.com/cloudquery/cloudquery/blob/bfe4714dbac5feeffe47ca8cc7ebcf3bf0d6d008/plugins/source/test/resources/plugin/plugin_fips.go#L26

where we append the `fipsEnabled` tag when configuring the build targets, similar to other build options

Reply:
Implemented it in [c83e36a](https://github.com/cloudquery/plugin-sdk/pull/2129/commits/c83e36ab792d44f5b82b2ded0460d457b77318f6)